### ISSUE TITLE
Create stream queues

### ISSUE THREAD

**Issue by herme5** [mbroadst/qamqp#91]
Hello,

Does this client support stream queues?

I read on the [RabbitMQ stream](https://www.rabbitmq.com/docs/streams) documentation that:

> "_An AMQP 0.9.1 client library that can specify [optional queue and consumer arguments](https://www.rabbitmq.com/docs/queues#optional-arguments) will be able to use streams as regular AMQP 0.9.1 queues. Just like queues, streams have to be declared first._"

It seems to be the case in qamqp so I tried the following:

```cpp
class StreamReceiver : public QObject
{
    Q_OBJECT
public:
    StreamReceiver(QObject *parent = 0) : QObject(parent) {
      m_client.setAutoReconnect(true);
    }

public Q_SLOTS:
    void start() {
        connect(&m_client, SIGNAL(connected()), this, SLOT(clientConnected()));
        m_client.connectToHost();
    }

private Q_SLOTS:
    void clientConnected() {
        QAmqpQueue *queue = m_client.createQueue("my-stream");
        disconnect(queue, 0, 0, 0); // in case this is a reconnect
        connect(queue, SIGNAL(declared()), this, SLOT(queueDeclared()));

        QAmqpTable args;
        args["x-queue-type"] = "stream";
        queue->declare(QAmqpQueue::Durable, args);
    }

    void queueDeclared() {
        QAmqpQueue *queue = qobject_cast<QAmqpQueue*>(sender());
        if (!queue)
            return;

        connect(queue, SIGNAL(messageReceived()), this, SLOT(messageReceived()));
        queue->consume();
        qDebug() << " [*] Waiting for messages. To exit press CTRL+C";
    }

    void messageReceived() {
        QAmqpQueue *queue = qobject_cast<QAmqpQueue*>(sender());
        if (!queue)
            return;

        QAmqpMessage message = queue->dequeue();
        qDebug() << " [x] Received " << message.payload();
    }

private:
    QAmqpClient m_client;
};

int main(int argc, char **argv)
{
    QCoreApplication app(argc, argv);
    StreamReceiver receiver;
    receiver.start();
    return app.exec();
}
```

TLDR: Queue `my-stream` is declared with just the `durable` option (no `autodelete` and no `exclusive`) and is configured with a `x-queue-type` argument set to `stream`.

---

## Test conditions

macOS Sequoia 15.1.1
Qt 5.15 for Desktop macOS
qamqp 0.6.0

Scenario: the client is responsible for creating the queue and there is no queue named "my-stream" that have been created beforehand.

## Bug

When executing the code, the queue never get declared (`queueDeclared()` slot is never called).
From the broker perspective, on the RabbitMQ management UI, the queue does not appear anywhere. From what I saw, stream queues appear alongside with standard ones in the queue tab.

## Expectation

The stream queue should be **created** and **declared** in the client so that consuming can start.

**Comment by herme5** [comment]
Nevermind I got it to work now with no change.
